examples/advanced_operations/add_responsive_search_ad_full.py

```diff
@@ -0,0 +1,574 @@
+            f"{geo_target_constant._COUNTRY_CODE}, "
+            f"{geo_target_constant.target_type}, "
+            f"{geo_target_constant.status.name}) "
+            f"is found in _LOCALE ({suggestion._LOCALE}) "
+            f"with reach ({suggestion.reach}) "
+            f"from search term ({suggestion.search_term})."
+        )
+        # Create the campaign criterion for location targeting.
+        campaign_criterion_operation = client.get_type("CampaignCriterionOperation")
+        campaign_criterion = campaign_criterion_operation.create
+        campaign_criterion.campaign = campaign_service.campaign_path(
+            customer_id, campaign_id
+        )
+        campaign_criterion.location.geo_target_constant = (
+            geo_target_constant.resource_name
+        )
+        operations.append(campaign_criterion_operation)
+
+    campaign_criterion_service = client.get_service("CampaignCriterionService")
+    campaign_criterion_response = campaign_criterion_service.mutate_campaign_criteria(
+        customer_id=customer_id, operations=[*operations]
+    )
+
+    for result in campaign_criterion_response.results:
+        print(f'Added campaign criterion "{result.resource_name}".')
+
+    return
+
+
+def add_images(client, customer_id, campaign_id):
+    # Step 6.1 - Add Image Asset.
+
+    # Download image from URL.
+    url = "https://img.cppng.com/download/2020-06/66912-logo-now-google-plus-search-free-transparent-image-hd.png"
+    image_content = requests.get(url).content
+
+    asset_service = client.get_service("AssetService")
+    asset_operation = client.get_type("AssetOperation")
+    asset = asset_operation.create
+    asset.type_ = client.enums.AssetTypeEnum.IMAGE
+
+    # Data field is the raw bytes data of an image.
+    asset.image_asset.data = image_content
+    asset.image_asset.file_size = len(image_content)
+
+    # MIME type of the image (IMAGE_JPEG, IMAGE_PNG, etc.).
+    # See more types on the link below.
+    # https://developers.google.com/google-ads/api/reference/rpc/v11/MimeTypeEnum.MimeType
+    asset.image_asset.mime_type = client.enums.MimeTypeEnum.IMAGE_PNG
+    # Use your favorite image library to determine dimensions
+    asset.image_asset.full_size.height_pixels = 1200
+    asset.image_asset.full_size.width_pixels = 1200
+    asset.image_asset.full_size.url = url
+    # Provide a unique friendly name to identify your asset.
+    # When there is an existing image asset with the same content but a different
+    # name, the new name will be dropped silently.
+    asset.name = f"Testing Image via API {uuid.uuid4()}"
+
+    mutate_asset_response = asset_service.mutate_assets(
+        customer_id=customer_id, operations=[asset_operation]
+    )
+    print("Uploaded file(s):")
+    for row in mutate_asset_response.results:
+        print(f"\tResource name: {row.resource_name}")
+
+    image_asset_id = mutate_asset_response.results[0].resource_name.split("/")[-1]
```

Comment:
No need to split out the ID here, you can just reuse the resource name as-is on line 507.

Reply:
TY!